pkg/services/store/entity/key.go

```diff
@@ -14,35 +14,49 @@ type Key struct {
 }
 
 func ParseKey(key string) (*Key, error) {
-	// /<group>/<resource>/<namespace>(/<name>(/<subresource>))
+
+	// Everything after resource is optional
+	// /<group>/<resource>/[<namespace>]/[<name>]/[<subresource>]
 	parts := strings.SplitN(key, "/", 6)
-	if len(parts) < 4 {
-		return nil, fmt.Errorf("invalid key (expecting at least 3 parts): %s", key)
+	if len(parts) < 3 {
```

Comment:
Unrelated to this PR, do you know if it's expected for `subresource` to contain forward slashes?

Otherwise, I'm wondering why we didn't go with `strings.Split` instead and validate that we don't have more than 6 parts.

Reply:
I don't think subresource is allowed to have forward slashes. 

I agree SplitN is more liberal here than we perhaps need to be. I would like to wait for Dan to return for making the switch to Split.

Reply:
It's an interesting question, looking at https://github.com/kubernetes/apiserver/blob/v0.29.2/pkg/endpoints/request/requestinfo.go it seems that (at least for RequestInfo) they don't treat additional url parts after the subresource as errors, they leave them in Parts but only take the first segment as Subresource.